src/universal_transfer_operator/data_providers/constants.py

```diff
@@ -0,0 +1,71 @@
+from enum import Enum
+
+
+class SupportedDataProvider(Enum):
+    """
+    List of all the supported DataProviders.
+    """
+
+    LOCAL = "local"
+    S3 = "s3"
+    FTP = "ftp"
+    HDFS = "hdfs"
+    SFTP = "sftp"
+    GCS = "gcs"
+    MYSQL = "mysql"
+    POSTGRES = "postgres"
+    BIGQUERY = "bigquery"
+    REDSHIFT = "redshift"
+
+
+class Database(Enum):
+    """
+    List of all the supported Databases DataProviders
+    """
+
+    POSTGRES = "postgres"
+    POSTGRESQL = "postgres"
+    SQLITE = "sqlite"
+    BIGQUERY = "bigquery"
+    SNOWFLAKE = "snowflake"
+
+
+class FileSystem(Enum):
+    """
+    List of all the supported FileSystem DataProviders
+    """
+
+    LOCAL = "local"
+    GCS = "gcs"  # Google Cloud Storage
+    S3 = "s3"  # Amazon S3
+
+
+class API(Enum):
+    """
+    List of all the supported API DataProviders
+    """
+
```

Comment:
What type of API do we want to support? In the past I heard from the Data team they'd be interested in using Salesforce API, Stripe, Prometheus/Thanos, and Splunk, for instance. Would we like to offer more specific API interfaces or generic ones?

Reply:
We can let users define the APIs in connection and we can support `GET`, `POST` and `PATCH` given source dataset and destination. Various authentication mechanisms like basic, OAuth, jwt token etc can be passed as part of the connection as well.